### PR TITLE
NRF53: add support for RPMSG HCI

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary

- boards/nrf5340-dk: add procfs support
- arch/nrf53: add RPMSG HCI controller support
- boards/nrf5340-dk: add RPMSG HCI controller support and examples

## Impact
BLE available to the app core.
## Testing
nrf5340-dk btsak and nimble on the app core

### CODE REVIEW

**Comment by raiden00pl** [comment]
depends on https://github.com/apache/nuttx/pull/8794

File: boards/arm/nrf53/nrf5340-dk/src/nrf53_bringup.c [hunk 1]
```diff
@@ -32,21 +33,122 @@
...
+{
+  struct bt_driver_s *bt_dev = NULL;
+  int                 ret    = OK;
+
+#ifdef CONFIG_BLUETOOTH_RPMSG
+  bt_dev = rpmsghci_register("appcore", "bthci");
+  if (bt_dev == NULL)
+    {
+      syslog(LOG_ERR, "ERROR: rpmsghci_register() failed: %d\n", -errno);
+      goto errout;
```
Comment (xiaoxiang781216) [inline boards/arm/nrf53/nrf5340-dk/src/nrf53_bringup.c:77]:
```suggestion
  return -ENOMEM;
```
and remove errout label

File: arch/arm/src/nrf53/nrf53_sdc.h [hunk 2]
```diff
@@ -56,6 +56,9 @@ extern "C"
  ****************************************************************************/
 
 int nrf53_sdc_initialize(void);
+#ifdef CONFIG_BLUETOOTH_RPMSG_SERVER
+int nrf53_rpmsghci_server_initialize(FAR const char *name);
```
Comment (xiaoxiang781216) [inline arch/arm/src/nrf53/nrf53_sdc.h:60]: remove all FAR from arch/ and board/ files

File: arch/arm/src/nrf53/nrf53_sdc.c [hunk 3]
```diff
@@ -913,9 +922,42 @@ int nrf53_sdc_initialize(void)
...
+{
+  int ret = OK;
+
+  /* Initialize SDC */
+
+  ret = nrf53_sdc_initialize();
+  if (ret < 0)
+    {
+      wlerr("nrf53_sdc_initialize error: %d\n", ret);
+      goto errout;
```
Comment (xiaoxiang781216) [inline arch/arm/src/nrf53/nrf53_sdc.c:949]: return ret and remove errout label

File: boards/arm/nrf53/nrf5340-dk/src/nrf53_bringup.c (hunk 1 above)
Comment (raiden00pl) [reply boards/arm/nrf53/nrf5340-dk/src/nrf53_bringup.c]: done

**Comment by raiden00pl** [comment]
depends on https://github.com/apache/nuttx/pull/8793 (https://github.com/apache/nuttx/pull/8793/commits/2710d8749e2bd343a57364cc99cc9e8f0e4d1d06)

**Comment by raiden00pl** [comment]
This PR pass CI now

**Review by jerpelea — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**